lib/UseStrictPlugin.js

```diff
@@ -2,19 +2,15 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var ConstDependency = require("./dependencies/ConstDependency");
+const ConstDependency = require("./dependencies/ConstDependency");
+const NullFactory = require("./NullFactory");
 
-var NullFactory = require("./NullFactory");
-
-function UseStrictPlugin() {}
-module.exports = UseStrictPlugin;
-
-UseStrictPlugin.prototype.apply = function(compiler) {
-	compiler.plugin("compilation", function(compilation, params) {
-		params.normalModuleFactory.plugin("parser", function(parser) {
-			parser.plugin("program", function(ast) {
-				var firstNode = ast.body[0];
-				var dep;
+class UseStrictPlugin {
+	apply(compiler) {
+		compiler.plugin("compilation", (compilation, params) => {
+			params.normalModuleFactory.plugin("program", (ast) => {
+				let firstNode = ast.body[0];
```

Comment:
I have a specific opinion about `const` and `let`. 

To me `const` should only be for scoping to `requires`/`imports` or for non Object primitives. Because you can truly guarantee their values will not change. Thus Object based primitives I leave as let as to not give someone the false sense of security that the values or properties inside are immutable.